### PR TITLE
July30th review

### PR DESCRIPTION
**Pull request by flametail9**
Corrections based on July 30th review session

### CODE REVIEW

File: Mat_test.cpp [hunk 1]
```diff
@@ -55,6 +55,11 @@ int main (){
     fprintf(outFile, "m.copy(newMat)\n");
     newMat.print(outFile);
 
+    Mat<float> outfloat;
+    outfloat = x.copy<float>();
```
Comment (anuranbaka) [inline Mat_test.cpp:59]: does x.copy(outfloat) work

File: Mat_test.cpp [hunk 2]
```diff
@@ -264,6 +269,21 @@ int main (){
...
     if(y.all()) fprintf(outFile, "All elements are true!\n");
     else fprintf(outFile, "Nope, at least one is false\n");
+    fprintf(outFile, "Checking if any elements of x are greater than 4:\n");
+    if((x > 4).any()) fprintf(outFile, "Some are greater!\n");
+    else fprintf(outFile, "Nope, none are\n");
+    fprintf(outFile, "Checking if all elements of y are less than 20:\n");
+    if((y < 20).all()) fprintf(outFile, "All elements are less than 20!\n");
+    else fprintf(outFile, "Nope, at least one is greater\n");
+    fprintf(outFile, "Checking if any elements of y are equal to 3:\n");
+    if((y == 3).any()) fprintf(outFile, "At least one is 3!\n");
```
Comment (anuranbaka) [inline Mat_test.cpp:279]: add tests for all and any on empty matricies
